File: Editorial/3761_Minimum_Absolute_Distance_Between_Mirror_Pairs/Go/1_Approach_One_time_Traversal.go
func minMirrorPairDistance(nums []int) int {
	reverseNum := func(x int) int {
		y := 0
		for x > 0 {
			y = y*10 + x%10
			x /= 10
		}
		return y
	}

	prev := make(map[int]int)
	n := len(nums)
	ans := n + 1

	for i, x := range nums {
		if idx, exists := prev[x]; exists {
			if i-idx < ans {
				ans = i - idx
			}
		}
		prev[reverseNum(x)] = i
	}

	if ans == n+1 {
		return -1
	}
	return ans
}

File: Editorial/3701-3800/3761_Minimum_Absolute_Distance_Between_Mirror_Pairs/TypeScript/1_Approach_One_time_Traversal.ts
function minMirrorPairDistance(nums: number[]): number {
    const reverseNum = (x: number): number => {
        let y = 0;
        while (x > 0) {
            y = y * 10 + (x % 10);
            x = Math.floor(x / 10);
        }
        return y;
    };

    const prev = new Map<number, number>();
    const n = nums.length;
    let ans = n + 1;

    for (let i = 0; i < n; i++) {
        const x = nums[i];
        if (prev.has(x)) {
            ans = Math.min(ans, i - prev.get(x)!);
        }
        prev.set(reverseNum(x), i);
    }

    return ans === n + 1 ? -1 : ans;
}
